sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowTablesSuite.scala

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command
+
+import org.apache.spark.sql.{QueryTest, Row}
+import org.apache.spark.sql.catalyst.analysis.{AnalysisTest, UnresolvedNamespace}
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser.parsePlan
+import org.apache.spark.sql.catalyst.plans.logical.{ShowTables, ShowTableStatement}
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.{BooleanType, StringType, StructType}
+
+trait ShowTablesSuite extends QueryTest with SharedSparkSession with AnalysisTest {
+  protected def catalog: String
+  protected def defaultUsing: String
+  case class ShowRow(namespace: String, table: String, isTemporary: Boolean)
+  protected def getRows(showRows: Seq[ShowRow]): Seq[Row]
+  // Gets the schema of `SHOW TABLES`
+  protected def showSchema: StructType
+
+  protected def runShowTablesSql(sqlText: String, expected: Seq[ShowRow]): Unit = {
+    val df = spark.sql(sqlText)
+    assert(df.schema === showSchema)
+    assert(df.collect() === getRows(expected))
+  }
+
+  protected def withSourceViews(f: => Unit): Unit = {
+    withTable("source", "source2") {
+      val df = spark.createDataFrame(Seq((1L, "a"), (2L, "b"), (3L, "c"))).toDF("id", "data")
+      df.createOrReplaceTempView("source")
+      val df2 = spark.createDataFrame(Seq((4L, "d"), (5L, "e"), (6L, "f"))).toDF("id", "data")
+      df2.createOrReplaceTempView("source2")
+      f
+    }
+  }
+
+  test("show tables") {
```

Comment:
Can we put the parser tests in `ShowTablesParserSuite`? It doesn't involve data source and doesn't need to be run twice with DS v1 and v2.

Reply:
Do you mean in a separate file or a class in the same file?

Reply:
In a separate file is better.